skops/card/tests/test_card.py

```diff
@@ -1887,3 +1887,20 @@ def test_toc_with_invisible_section(self, card):
         ]
 
         assert toc == "\n".join(exptected_toc)
+
+
+class TestCardSaveWithPlots:
+    def test_copy_plots(self, destination_path, model_card):
+        import matplotlib.pyplot as plt
+
+        plt.plot([4, 5, 6, 7])
+        plt.savefig(Path(destination_path) / "fig1.png")
+        model_card = model_card.add_plot(fig1="fig1.png")
```

Comment:
The plot path here is already incorrect, is it not? Because you save the figure at `Path(destination_path) / "fig1.png"` but reference `"fig1.png"`.
(btw we may want to verify in a future PR that a plot exists when calling `add_plot`)

Reply:
I was basing this off of this `TestAddPlot` test https://github.com/skops-dev/skops/blob/3ab308bd9beab44e816272790aaa64e04d3f6910/skops/card/tests/test_card.py#L1101

Reply:
So the point of this PR is what was described in #300. A passage from there that fits this situation is:

> Now I can work around this by changing my code to

```python
model_card = card.Card(...)
fig.savefig(hub_dir / "my_plot.png")  # <= use abs path here
model_card.add_plot(**{"My beautiful plot": "my_plot.png"})  # <= use rel path here
```

> but this is an ugly gotcha. Also, even though it will work correctly on the Hub, when I render the model card locally to preview it, it will fail because the paths mismatch.

So the test you're referring to does exactly that, using a mix of relative and absolute path, which is exactly what we want to avoid with the new feature. That's why I think it's important for this test to not use that pattern. We could adjust the other tests as well, but that's really not so important there, since what they're testing is not concerned with the paths. I hope that clears it up.